subject_files_status_for_dpdash2.py

```diff
@@ -43,22 +57,43 @@ def get_mri_status():
     """
     
 
-    interview_date=get_value(timepoint,'chrmri_entry_date')
+    interview_date=get_value_from_dict(dict1,timepoint,'chrmri_entry_date')
     if interview_date=='':
         return {'mri_score':'', 'mri_data':'', 'mri_protocol':'', 'mri_date':'', 'mri_missing':''}
 
-    if get_value(timepoint,'chrmri_missing')=='1':
-        missing_code=get_value(timepoint,'chrmri_missing_spec')
+    if get_value_from_dict(dict1,timepoint,'chrmri_missing')=='1':
+        missing_code=get_value_from_dict(dict1,timepoint,'chrmri_missing_spec')
         return {'mri_score':'', 'mri_data':'', 'mri_protocol':'', 'mri_date':interview_date, 'mri_missing':missing_code}
 
     scan_minus_consent=str_date_minus_str_date(consent_date,interview_date)
     days_since_scan=str_date_minus_str_date(interview_date,today)
     
 
     try:
-        for s,row in df_mri.loc[subject].iterrows():
-            if timepoint in row['timepoint_text']:
-                break
+        # get information using subject ID and entry_date
+        df_subject = df_mri.loc[subject]
+
+        # restrict to the rows with run sheet information
+        df_subject = df_subject[~df_subject.file_loc.isnull()]
+        if len(df_subject) == 0:
+            raise KeyError
+
+        # match the row based on the interview date
+        df_subject_scan = df_subject.set_index(
+                'entry_date').loc[[interview_date]]
+        if test:
+            print(df_subject_scan)
+
+        if len(df_subject_scan) == 1:
+            row = df_subject_scan.iloc[0]
+        elif len(df_subject_scan) == 0:
+            raise AttributeError
+        else:
+            # multiple lines for the subject and scan date
+            print(f'More than one matched information for {subject} '
+                  f'{timepoint} in the mri_all_db.csv. '
+                  f'Using the first row.')
+            row = df_subject_scan.iloc[0]
```

Comment:
Lines 73-96 are the substantial work in this PR. Kevin defines `row` in these lines that are later used to obtain score line 75:
> score=int(row['mriqc_int'])

Reply:
My comments are the following:

1. @kcho , if you manually raise `KeyError`, `AttributeError`, are those caught by the exception in line 98? Remember, this program runs in a continuous loop and we cannot have one subject halt iteration over other subjects.

2. The print statement--where do you plan to monitor it really? This program runs as a cron job that only emails me the logs every night. Are you planning to run `subject_files_status_for_dpdash2.py` manually once in a while? If not, can we omit this print statement and keep the text as a comment in code? We don't have any other asymmetric print statement in this program.

Otherwise, as long as you have tested this new block, it can be integrated into this program.

Reply:
1. Yes.
2. I'll remove the print statement.

Reply:
Removing print https://github.com/AMP-SCZ/utility/pull/108